### DISCUSSION TITLE
Pinia in lib mode SSR

### DISCUSSION THREAD

**Discussion by ansbar** [vuejs/pinia#1799] [Help and Questions]
Hi, we're using a Pinia store in lib mode and using the lib in a SSR project.
The lib is internal only and all consumers of the lib already uses Pinia in their projects. 
We've had some difficulties making the store work in the lib on the server but by externalising Pinia and using a prop with $pinia it seems to be rolling now.

However, using the store in the main project isn't working 100% yet.

I have two questions.
1. I want to access my lib store in my main project. Can I get a hold of the $pinia instance in setup somehow?
2. Is this (an unusual but) proper way of utilising Pinia?

```vue
 // Main Project
<script setup lang="ts">
  import myLib from "@node_modules/lib"
  import useMyStore from "@node_modules/lib/src/stores/myStore"
  
  const myStore = useMyStore() // [Vue warn]: injection "Symbol(pinia)" not found.
  const myStore = useStore($pinia) // Cannot find name $pinia

  console.log(myStore.myAction)
</script>
<template>  
  <myLib :pinia="$pinia"/> <!-- Works -->
</template>
```

// My Library
myLib.vue
```vue
<script setup lang="ts">
   import { Pinia } from "pinia"
   import useMyStore from "@/stores/MyStore"
   
  const p = defineProps({
    pinia: {
      type: Object as PropType<Pinia>,
      default: () => ({}),
    },
  })
  
  const myStore = useMyStore(p.pinia)  
  console.log(myStore.myState) // Working
</script>
```
// Index lib file
```ts
import myLib from "@/myLib.vue"
import { useMyStore } from "./stores/myStore"
export { myLib, useMyStore }
```

**Comment by ansbar** [comment]
@posva do you know if question 1 is achievable?
(I want to access my lib store in my main project. Can I get a hold of the $pinia instance in setup somehow?)

**Reply by posva** [reply]
when creating a library that uses pinia, it is expected from the application consuming it to add pinia itself. **The library shouldn't add pinia to the app**.
